source/documentation/monitoring-an-app/application-metrics.md

```diff
@@ -1,107 +1,130 @@
 
-#### NetworkPolicy for Monitoring Namespace
+```
+kubectl -n <namespace> get svc rails-app-service -o=jsonpath={.spec.ports[0].name}
+```
 
-By default, all connections from outside a namespace are blocked, therefore a network policy is required for the `monitoring` namespace to be able to connect into an application namespace to scrape the metrics endpoint.
+As you can see, the name of the port we're exposing is `http`.
 
-Example:
+Create and apply your service monitor `<application>-serviceMonitor.yaml`, as below:
 
 ```yaml
-kind: NetworkPolicy
-apiVersion: networking.k8s.io/v1
-metadata:
-  name: allow-prometheus-scraping
-  namespace: my-app-namespace
-spec:
-  podSelector:
-    matchLabels:
-      app: my-app
-  policyTypes:
-  - Ingress
-  ingress:
-  - from:
-    - namespaceSelector:
-        matchLabels:
-          component: monitoring
-```
-You can view your current NetworkPolices with the following command:
+   apiVersion: monitoring.coreos.com/v1
+   kind: ServiceMonitor
+   metadata:
+     name: rails-app-service
+   spec:
+     selector:
+       matchLabels:
+         app: rails-app-service
+     endpoints:
+     - port: http # this is the port name you grabbed from your running service
+       interval: 15s
+   ```
+
+This will tell Prometheus to go and scrape that endpoint every 15 seconds and store any exposed metrics.
+
+#### Add a NetworkPolicy resource
+ 
+The Prometheus server is in the 'monitoring' namespace, but by default, any network connections from outside your application's namespace will be blocked. So, to allow prometheus to scrape your application's `/metrics` endpoint, we need to add a network policy to allow connections from the monitoring namespace.
+
+Create and apply a new resource `<application>-networkPolicy.yaml`, as below:
+
+```yaml
+   kind: NetworkPolicy
+   apiVersion: networking.k8s.io/v1
+   metadata:
+     name: allow-prometheus-scraping
+     namespace: my-app-namespace
+   spec:
+     podSelector:
+       matchLabels:
+         app: rails-app
+     policyTypes:                                                                                                                                                         
+     - Ingress
+     ingress:
+     - from:
+       - namespaceSelector:
+           matchLabels:
+             component: monitoring
+   ```
+
+#### Querying metrics
+
+We can now query out `/metric` endpoint using the Cloud Platform Prometheus.
+
+Head to [Cloud Platform Prometheus](https://prometheus.cloud-platform.service.justice.gov.uk/graph) and use the following promql query to view the application latency (remembering to change the namespace value):
 
-```sh
-kubectl -n <namespace> get networkpolicies
 ```
+http_server_request_duration_seconds_sum{namespace="my-namespace"}
+```
+
+The output will be something like:
+![Image of prometheus output](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/prometheus.png)
```

Comment:
This image doesn't display. Also, you should use a relative path like `(../images/multi-container-k8s.png)`